lib/kaffe/group_member/worker/worker_manager.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Kaffe.WorkerManager do
+  @moduledoc """
+  This is the main process for launching group members and workers.
+
+  The process begins by starting the client connection to Kafka. Then, group
+  members are created for each of the configured topics.
+
+  This is also the place where subscribers get the PID for a Worker.
+  """
+  
+  use GenServer
+
+  alias Kaffe.WorkerSupervisor
+
+  require Logger
+
+  def start_link do
+    Logger.debug "Starting worker manager"
+    GenServer.start_link(__MODULE__, [self()])  
+  end
+
+  def worker_for(pid, partition) do
+    GenServer.call(pid, {:worker_for, partition})
+  end
+
+  ## Callbacks
+
+  def init([supervisor_pid]) do
+    Logger.info "event#starting=#{inspect self()}"
+
+    worker_table = :ets.new(:workers, [:set, :protected])
```

Comment:
This is a broker for workers. It knows the partition to worker assignment. I think keeping the supervisors simple minimizes the chance of them crashing which enhances overall reliability.